### ISSUE TITLE
Loading of empty delivery Days Window take long

### ISSUE THREAD

**Issue by metas-mk** [metasfresh/metasfresh-webui-frontend#847]
### Type of issue
Feature request
### Current behavior
Currently, the first opening and reload after refresh takes very long for window/540110 delivery days. Also when refreshing/ reloading it shows weird behavior.
### Expected behavior
Something that is better for the user. Let's discuss ;-)

![image](https://user-images.githubusercontent.com/13365687/26881999-64db7720-4b99-11e7-95c8-1a4858c1cb4a.png)

**Comment by cadavre** [comment]
It is pure backend time (12.30s) which frontend is waiting to get response. Only thing frontend can do is to add some spinner, maybe with "Loading data..." text, into container of GridView. I'll need more feedback from @damianprzygodzki how can we achieve this and how to make it in generic way.

**Comment by teosarca** [comment]
@cadavre  True, that would be the best.
But also, observe that after refresh, not even the layout is shown until the data is fetched,
So frontend should always show the layout ASAP.

Remark that on first time when i am loading it, it works exactly how it's needed. Layout is displayed ASAP. And when data is available, the rows are displayed.

**Comment by damianprzygodzki** [comment]
@cadavre I think that the best thing to do for gridview without data is to add some initial text in place of "There are no detail rows" something like "Loading the data...".

@teosarca The case is that for performance issue we allow to show only gridview, when and data and layout are ready to do not re-render whole view few times due to component dependencies in data (QuickActions, Table (size, orderBy, queryHitLimit etc)). We can remove that, but it is change in assumption, may cause lot of problems, so refactor and performance research needed. We should postpone that.

**Comment by teosarca** [comment]
Hi @damianprzygodzki 

IMHO, the frontend shall be as fluent as possible.
The case of "when loading the data the endpoint fails" it's an exception and we shall not optimize for that.
Usually, "retrieving the data" works without any error.

More, atm, the behaviour is not consistent.

Case 1: Here is how it looks when the view is loaded first time, i.e. you can see it by clicking on https://w101.metasfresh.com:8443/window/540110

![image](https://user-images.githubusercontent.com/1244701/26915363-64501546-4c2e-11e7-9da8-839265be2794.png)


Case 2: Here is how it looks when i am refreshing the page

![image](https://user-images.githubusercontent.com/1244701/26915411-91f2ef50-4c2e-11e7-820b-49f3f001b6c7.png)

**Comment by damianprzygodzki** [comment]
> IMHO, the frontend shall be as fluent as possible.

Fluent with bad performance is impossible :)

> The case of "when loading the data the endpoint fails" it's an exception and we shall not optimize for that.

We do not optimize for that.

And in reference to other screeshots. I am aware about all that things. And i can just copy and paste my last message.

> We can remove that (...) may cause lot of problems. (...) We should postpone that (move to the farthest milestone we can).

**Comment by teosarca** [comment]
ok, let's talk when u have time because to be honest I don't understand

**Comment by cadavre** [comment]
We found out that this is caused by already generated `viewId` that is being used (12s) instead of empty placeholder API call (50ms) which returned meta-data that is required in frontend.

On call we made assumption to make "Minimum Viable Solution" to solve this for now:

1. If it is possible and won't make top loading bar flickering – add loader bar for **each** API call made.

2. But if it will make this loading bar flicker in unwanted way – we add loader bar moving **additionally only for** loading data in gridview.

With no.1 we're waiting for @damianprzygodzki to make a live preview or screen movie or GIF showing how it works and judge if it's flickering or not.

Reminder: this is not high priority.
